src/server/endpoints/home-screen-endpoints.js

```diff
@@ -1,13 +1,67 @@
 'use strict';
 
 const videosDb = require('../collections/videos-db');
+const tokensDb = require('../collections/tokens-db');
 
 function getHomeInfos(req, res) {
   videosDb.findVideoInfo(req.params.videoId, (videoInfos) => {
     if (videoInfos._id === undefined) {
       res.status(404).json({'error': 'not found'});
       return;
     }
+    let videoLikeNums = 0;
+    let videoDislikeNums = 0;
+    let token = req.get('Authorization');
+    let userId;
+
+    tokensDb.getToken(token, (userInfos) => {
+      userId = userInfos.userId;
+      videoInfos.videoDetails.clickedLike = false;
+      videoInfos.videoDetails.clickedDislike = false;
+
+      videoInfos.videoDetails.LikeStatus.forEach((user) => {
+        if (user.userId.toString() === userId.toString() && user.liked === true) {
+          videoInfos.videoDetails.clickedLike = true;
+        }
+        if (user.userId.toString() === userId.toString() && user.disliked === true) {
+          videoInfos.videoDetails.clickedDislike = true;
+        }
+      });
+      console.log('userliked:' + videoInfos.videoDetails.LikeStatus[2].liked);
+      res.status(200).json({
+        // videoId should be same with _id, here just for testing
+        // in your PC please change here
+        'videoId': '59eecd63abb2117ba6f42a15',
+        'videoUrl': 'http://nettuts.s3.amazonaws.com/763_sammyJSIntro/trailer_test.mp4',
+        'videoDetails': videoInfos.videoDetails,
+
+        'uploader': {
+          'name': 'uploader name',
+          'userId': '59e02b5d120d8b14a06816b6',
+          'avatar': 'https://encrypted-tbn0.gstatic.com/images?q=tbn:ANd9GcQLDxSdH8lLX-y9TJzLDWZPvoLexXrE8Ft5EAAWaZNyQHVM-yh-3A',
+          'subscribers': 1000,
+        },
+        'commentInfos': videoInfos.commentInfos,
+      });
+    });
+
+    videoInfos.videoDetails.LikeStatus.forEach((videoLikeStatus, index) => {
+      if (videoInfos.videoDetails.LikeStatus === 0) {
```

Comment:
- you dont need to iterate through the likes again
 - do not do anything out of the getToken callback